### PR TITLE
Document java docker build steps

### PR DESCRIPTION
**Pull request by adfoster-r7**

### CODE REVIEW

File: java/README.md [hunk 1]
```diff
@@ -3,13 +3,38 @@
...
+$ ls working_directory
+metasploit-framework
+metasploit-payloads
+```
+
+First build the Docker image:
+```
+cd working_directory/metasploit-payloads/docker
+docker build -t metasploit_payloads_builder .
+```
```
Comment (jmartin-r7) [inline java/README.md:26]: Should we add or `docker pull rapid7/msf-ubuntu-x64-meterpreter:latest`?

File: java/README.md [hunk 2]
```diff
@@ -3,13 +3,38 @@
 1. Install Maven and Java, this will depend on your OS
 1. Download the [Android SDK](https://developer.android.com/sdk/index.html)
 1. Install Android SDK Platforms 3, 10 and 19, and update the "Android SDK Tools" and "Android SDK Platform-tools"
-1. Compile the Android and Java Meterpreter, which deploys to the ../metasploit-frameworks folder
+1. Compile the Android and Java Meterpreter, which deploys to the `../metasploit-frameworks` folder
```
Comment (timwr) [inline java/README.md:6]: `../metasploit-frameworks` or `../metasploit-framework`?

File: java/README.md (hunk 1 above)
Comment (timwr) [reply java/README.md:26]: I agree, maybe pull the existing image might be simpler/quicker

File: java/README.md [hunk 3]
```diff
@@ -3,13 +3,38 @@
...
+
+Next run the `metasploit_payloads_builder` image as a container and mount the `working_directory`.
+This interactive shell will allow you to compile the Android and Java Meterpreter, and deploy
+to the `../metasploit-frameworks` folder as normal:
+```
+cd working_directory
+docker run --rm  -it -w $(pwd) -v $(pwd):$(pwd) metasploit_payloads_builder /bin/bash
+
+cd metasploit-payloads/java
+mvn package -Dandroid.sdk.path=/usr/local/android-sdk -Dandroid.release=true -P deploy
```
Comment (timwr) [inline java/README.md:36]: Alternatively just `make android`?

**Pull request merged**